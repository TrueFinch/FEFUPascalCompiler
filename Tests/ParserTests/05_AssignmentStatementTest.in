program p;
var
    a, b: float;
begin
    a := 1;
    b := 2;
    
    a += b;
    b += a;
    
    b -= a;
    a -= b;
    
    a *= b;
    b *= a;
    
    a /= 100;
    b /= 100;
    writeln(a, b);
end.