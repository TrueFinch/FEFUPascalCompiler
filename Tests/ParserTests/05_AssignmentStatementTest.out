                                              p                                                
                                              ┼                                                
                                          MainBlock                                            
       ┌──────────────────────────────────────┴───────┐                                        
      var                                     CompoundStatement                                
       ┼         ┌─────┬─────┬─────┬─────┬─────┬─────┬┴────┬──────┬───────┬───────────┐        
 SimpleVarDecl   :=    :=    +=    +=    -=    -=    *=    *=     /=      /=    FunctionCall   
 ┌──┬──┴─┐      ┌┴─┐  ┌┴─┐  ┌┴─┐  ┌┴─┐  ┌┴─┐  ┌┴─┐  ┌┴─┐  ┌┴─┐  ┌─┴─┐   ┌─┴─┐      ┌──┴──┬──┐  
 a  b  float    a  1  b  2  a  b  b  a  b  a  a  b  a  b  b  a  a  100  b  100  writeln  a  b  